src/lib/swapper/swappers/ThorchainSwapper/getThorTradeQuote/getTradeQuote.ts

```diff
@@ -143,369 +96,16 @@ export const getThorTradeQuote = async (
-          }),
-        )
-
-      // otherwise, return all that succeeded
-      return Ok(routes)
-    }
-
-    case CHAIN_NAMESPACE.CosmosSdk: {
-      const sellAdapter = assertGetCosmosSdkChainAdapter(sellAsset.chainId)
-      const feeData = await sellAdapter.getFeeData({})
-
-      return Ok(
-        perRouteValues.map(
-          ({
-            source,
-            quote,
-            expectedAmountOutThorBaseUnit,
-            isStreaming,
-            estimatedExecutionTimeMs,
-            affiliateBps,
-          }): ThorTradeQuote => {
-            const rate = getRouteRate(expectedAmountOutThorBaseUnit)
-            const buyAmountBeforeFeesCryptoBaseUnit = getRouteBuyAmount(quote)
-
-            const buyAmountAfterFeesCryptoBaseUnit = convertPrecision({
-              value: expectedAmountOutThorBaseUnit,
-              inputExponent: THORCHAIN_FIXED_PRECISION,
-              outputExponent: buyAsset.precision,
-            }).toFixed()
-
-            const updatedMemo = addSlippageToMemo({
-              affiliateFeesThorBaseUnit: quote.fees.affiliate,
-              expectedAmountOutThorBaseUnit,
-              quotedMemo: quote.memo,
-              slippageBps: inputSlippageBps,
-              isStreaming,
-              chainId: sellAsset.chainId,
-              affiliateBps,
-              streamingInterval,
-            })
-
-            return {
-              id: uuid(),
-              memo: updatedMemo,
-              receiveAddress,
-              affiliateBps,
-              isStreaming,
-              rate,
-              steps: [
-                {
-                  estimatedExecutionTimeMs,
-                  rate,
-                  sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmountCryptoBaseUnit,
-                  buyAmountBeforeFeesCryptoBaseUnit,
-                  buyAmountAfterFeesCryptoBaseUnit,
-                  source,
-                  buyAsset,
-                  sellAsset,
-                  accountNumber,
-                  allowanceContract: '0x0', // not applicable to cosmos
-                  feeData: {
-                    networkFeeCryptoBaseUnit: feeData.fast.txFee,
-                    protocolFees: getProtocolFees(quote),
-                    chainSpecific: {
-                      estimatedGasCryptoBaseUnit: feeData.fast.chainSpecific.gasLimit,
-                    },
-                  },
-                },
-              ],
-            }
-          },
-        ),
-      )
-    }
-
+  switch (tradeType) {
+    case TradeType.L1ToL1:
+      return getL1quote(input, streamingInterval)
+    case TradeType.LongTailToL1:
+      return getLongtailToL1Quote(input, streamingInterval, assetsById)
+    case TradeType.LongTailToLongTail:
+      return Err(makeSwapErrorRight({ message: 'Not implemented yet' }))
+    case TradeType.L1ToLongTail:
+      return Err(makeSwapErrorRight({ message: 'Not implemented yet' }))
```

Comment:
we can fallthrough, if linter complains you can add a comment `// fallthrough` to silence the warning
```suggestion
    case TradeType.LongTailToLongTail:
    case TradeType.L1ToLongTail:
      return Err(makeSwapErrorRight({ message: 'Not implemented yet' }))
```

Reply:
https://github.com/shapeshift/web/pull/5638/commits/27a1c79332b85efd53dde708777cf2687aec5322